From 78629e74e1e046a9ae4945088e2f7afde6db88d6 Mon Sep 17 00:00:00 2001
From: Fridrich Strba <fridrich_strba@opensuse.org>
Date: Tue, 12 Mar 2024 11:34:52 +0100
Subject: Fix build with autoconf 2.72

This commit http://git.savannah.gnu.org/gitweb/
?p=autoconf.git;a=commitdiff;
h=cf09f48841b66fe76f606dd6018bb3a93242a7c9
changed the internal cache variable name and its content

The effects show later in message like
/tmp/ccyB6wS6.ltrans9.ltrans.o: in function `comphelper::DocPasswordHelper::decryptGpgSession(com::sun::star::uno::Sequence<com::sun::star::uno::Sequence<com::sun::star::beans::NamedValue> > const&)':
/home/abuild/rpmbuild/BUILD/libreoffice-24.2.1.2/comphelper/source/misc/docpasswordhelper.cxx:705:(.text+0x4fb7): undefined reference to `GpgME::Data::seek(long, int)'
/usr/bin/ld.bfd: /home/abuild/rpmbuild/BUILD/libreoffice-24.2.1.2/comphelper/source/misc/docpasswordhelper.cxx:716:(.text+0x5402): undefined reference to `GpgME::Data::seek(long, int)'
/usr/bin/ld.bfd: /tmp/ccyB6wS6.ltrans13.ltrans.o: in function `comphelper::OStorageHelper::CreateGpgPackageEncryptionData()':
/home/abuild/rpmbuild/BUILD/libreoffice-24.2.1.2/comphelper/source/misc/storagehelper.cxx:507:(.text+0x64e3): undefined reference to `GpgME::Data::seek(long, int)'
/usr/bin/ld.bfd: /home/abuild/rpmbuild/BUILD/libreoffice-24.2.1.2/comphelper/source/misc/storagehelper.cxx:520:(.text+0x6583): undefined reference to `GpgME::Data::seek(long, int)'
collect2: error: ld returned 1 exit status

Where the type 'off_t' at the time of inclusion of the header and
at the time of compilation of the library differ.

Change-Id: Ie0486dbc869e84f5fb2688473334cfe464abd570
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/164704
Tested-by: Jenkins
Reviewed-by: Stephan Bergmann <stephan.bergmann@allotropia.de>
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/164715
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/164764
Reviewed-by: Miklos Vajna <vmiklos@collabora.com>
Reviewed-by: Michael Meeks <michael.meeks@collabora.com>
Tested-by: Stephan Bergmann <stephan.bergmann@allotropia.de>
---
 configure.ac | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 8101a2b2e1dd..758e160400ba 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7197,7 +7197,9 @@ if test "$_os" != "WINNT"; then
 
     dnl Check for large file support
     AC_SYS_LARGEFILE
-    if test -n "$ac_cv_sys_file_offset_bits" -a "$ac_cv_sys_file_offset_bits" != "no"; then
+    if test -n "$ac_cv_sys_largefile_opts"  -a "$ac_cv_sys_largefile_opts" != "none needed" -a "$ac_cv_sys_largefile_opts" != "support not detected"; then
+        LFS_CFLAGS="$ac_cv_sys_largefile_opts"
+    elif test -n "$ac_cv_sys_file_offset_bits" -a "$ac_cv_sys_file_offset_bits" != "no"; then
         LFS_CFLAGS="-D_FILE_OFFSET_BITS=$ac_cv_sys_file_offset_bits"
     fi
     if test -n "$ac_cv_sys_large_files" -a "$ac_cv_sys_large_files" != "no"; then
-- 
cgit v1.2.3

